core/string/string_builder.h

```diff
@@ -36,13 +36,10 @@
 
 class StringBuilder {
 	uint32_t string_length = 0;
+	uint32_t capacity = 16;
+	uint32_t strings = 0;
 
-	Vector<String> strings;
-	Vector<const char *> c_strings;
-
-	// -1 means it's a Godot String
-	// a natural number means C string.
-	Vector<int32_t> appended_strings;
+	char32_t *buffer;
 
```

Comment:
Maybe you can just use a `LocalVector` instead of handling memory internally? This whole class seems much like a wrapper on `LocalVector`. :thinking:

Reply:
That would actually just increase the code size, and force me to double-check that every assumption holds for the specific method calls I use. And would be susceptible to future changes in `LocalVector`. I refer you to the story of how `left-pad` took down the internet. Not everything has to be a web of inter-dependencies.

At any rate the discussion is moot. It turns out there's another class called `StringBuffer` that does everything this naive implementation does, plus a couple of extras, achieving better performance on benchmarks. So an even smaller solution would be to just typedef `StringBuilder` to `StringBuffer`. Or remove use of the prior from the code completely.